src/lib/yacc/grammar.rs

```diff
@@ -309,7 +323,7 @@ impl YaccGrammar {
 
     /// Return the name of terminal `i` or `None` if it doesn't exist.
     pub fn term_name(&self, i: TIdx) -> Option<&str> {
-        self.term_names.get(usize::from(i)).map_or(None, |x| Some(x))
+        self.term_names.get(usize::from(i)).map_or(None, |x| x.as_ref().map_or(None, |y| Some(&y)))
```

Comment:
I'm not saying there isn't, but this is the best I came up with. It's a bit more complex than it first looks: it basically says "if the name is None return None, otherwise coerce the `String` to a `&str` and return".